File: problems/2022_09_12/0150/Solution.java
/*
 * Author: digmouse233
 * Time Complexity: O(n)
 * Space Complexity: O(n)
 */

public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();

        for (String token : tokens) {
            if (token.equals("+")) {
                stack.add(stack.pop() + stack.pop());
            } else if (token.equals("*")) {
                stack.add(stack.pop() * stack.pop());
            } else if (token.equals("-")) {
                int secNum = stack.pop();
                int firNum = stack.pop();
                stack.add(firNum - secNum);
            } else if (token.equals("/")) {
                int secNum = stack.pop();
                int firNum = stack.pop();
                stack.add(firNum / secNum);
            } else {
                stack.add(Integer.parseInt(token));
            }
        }

        return stack.peek();
    }
}

File: problems/2022_09_12/0150/Solution.cpp
#include <fstream>
#include <iostream>
#include<vector>
using namespace std;

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;
        for (int i=0; i<tokens.size(); i++){
            if (tokens[i] != "+" && tokens[i] != "-" && tokens[i] != "*" && tokens[i] != "/"){
                stk.push(stoi(tokens[i]));
            } 
            else {
                int num1 = stk.top();
                stk.pop();
                int num2 = stk.top();
                stk.pop();
                if(tokens[i]=="+"){
                    stk.push(num1 + num2);
                }
                else if(tokens[i]=="-"){
                    stk.push(num2 - num1);
                }
                else if(tokens[i]=="*"){
                    stk.push((long)num1 * (long)num2);
                }
                else if(tokens[i]=="/"){
                    stk.push(num2 / num1);
                }
            }
        }
        return stk.top();
    }
};